lib/ansible/modules/clustering/etcd3.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+import etcd3
+
+
+DOCUMENTATION = """
+---
+module: etcd3
+
+short_description:
+   - Queries etcd3 for key/value storage
+
+version_added: "2.4.2.0"
+
+description:
+   - Query etcd3 for key/value storage
+
+options:
+    key:
+        description:
+            - the key where the information is stored in the cluster
+        required: true
+    value:
+        description:
+            - the information stored
+        required: true
+    url:
+        description:
+            - the address to the cluster
+        required: true
+    state:
+        description:
+            - the state of the value for the key.
+        required: true
+author:
+    - Jean-Philippe Evrard (@evrardjp)
+"""
+
+EXAMPLES = """
+# Store a value "bar" under the key "foo" for a cluster located "http://localhost:2379"
+- etcd3:
+    key: "foo"
+    value: "baz3"
+    url: "http://localhost:2379"
+    state: "present"
+"""
+
+RETURN = '''
+message:
+    description: The output message that the sample module generates
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import etcd3
+
+
+def run_module():
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        key=dict(type='str', required=True),
+        value=dict(type='str', required=True),
+        host=dict(type='str', default='localhost'),
+        port=dict(type='int', default=2379),
+        state=dict(type='str', required=True)
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        #message=''
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    allowed_keys = ['host', 'port', 'ca_cert', 'cert_key', 'cert_cert', 'timeout', 'user', 'password']
+    client_params = {key: value for key, value in module.params.items() if key in allowed_keys}
+    try:
+        etcd = etcd3.client(**client_params)
+    except Exception as exp:
+        module.fail_json(msg='Cannot connect to etcd cluster. Exception {}'.format(module.params))
```

Comment:
It will be easy for debugging if we print `exp` exception to user.
Use `to_native(exp)`